nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -953,14 +962,21 @@ ngx_http_modsecurity_init(ngx_conf_t *cf)
     }
     *h = ngx_http_modsecurity_handler;
 
-    ngx_http_next_header_filter = ngx_http_top_header_filter;
-    ngx_http_top_header_filter = ngx_http_modsecurity_header_filter;
-
-    ngx_http_next_body_filter = ngx_http_top_body_filter;
-    ngx_http_top_body_filter = ngx_http_modsecurity_body_filter;
-
     ngx_memzero(&ngx_http_modsecurity_upstream, sizeof(ngx_http_upstream_t));
     ngx_http_modsecurity_upstream.cacheable = 1;
+
+    ngx_thread_pool_t* thread_pool = ngx_thread_pool_add(cf, &thread_pool_name);
+    if (thread_pool == NULL) {
```

Comment:
Should we add retries for some of the critical failures?

Reply:
none of the other similar modules do this, so I think we shouldn't. If this low level thing fails during config time, I think it's out of our control.